core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java

```diff
@@ -133,7 +125,13 @@ private void collectAndRemoveUncommittedData() throws IOException {
   }
 
   private void collectAndRemoveOldVersionData() throws IOException {
-    long deleteTimeLine = System.currentTimeMillis() - config.get(KV_DELETE_AFTER_TIME);
+    // If users use the deprecated configuration, we will give priority to the deprecated value,
+    // otherwise the new configuration and its default values will be used.
+    long dateTimeLineMillis = config.get(STORE_DELETE_AFTER_TIME);
+    if (config.get(KV_DELETE_AFTER_TIME).isPresent()) {
```

Comment:
New configuration option should override legacy configuration option, right?

Reply:
> New configuration option should override legacy configuration option, right?

For compatibility, if the user previously configured the old configuration, i think the old configuration should override the new one or its default value.

Reply:
Maybe you refer to other system how to handle the case like Spark. I feel this is weird.

Reply:
> For compatibility, if the user previously configured the old configuration, i think the old configuration should override the new one or its default value.

I would suggest this one. Or, if a user upgrades Gravitinio to a new version, the value of some configuration may change.

Reply:
I think new configuration should use legacy configuration as default value instead of a new default value for the compatibility. But if we set a new configuration, the new configuration will override the old configuration. 

I think we can look at other systems how to handle it. This is a common case. We should adopt a common to solve it.

Reply:
got it, thanks both, i will try to make some research.

Reply:
I have look at the spark handling deprecated config:

When get the new config, firstly read the new config value, then read the alternative old one, finally read the new config default value.

Take a fresh look at the problem scene:
A, the old deprecated config, and default value is a.
B, the new config, and default value is b.

For implementation, when the garbage collector call `get(B)`, we learn from spark and expect that:
1) If B is set, using B's value
2) If B is not set, A is set, using A's value (_be compatible with the deprecated config for users_)
3) If both B and A are not set, using B's default value

To implement the second expected item, we need a `fallback` method to get A's value, for example
```
new ConfigBuilder(STORE_DELETE_AFTER_TIME_KEY)
    .fallback(KV_DELETE_AFTER_TIME_KEY)

```
and in `readFrom` method to get the fallback config value (B is not set and A is set).

@qqqttt123 @yuqi1129  Is that OK?  WDYT?

Reply:
https://github.com/apache/incubator-uniffle/blob/master/common/src/main/java/org/apache/uniffle/common/config/FallbackKey.java